doc/BGP/BGP-router-id.md

```diff
@@ -0,0 +1,127 @@
+# BGP Router ID Explicitly Configured
+
+- [Revision](#revision)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Scope](#scope)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [High Level Design](#high-level-design)
+- [Config DB Enhancement](#config-db-enhancement)
+    - [DEVICE_METADATA](#device_metadata)
+
+### Revision
+
+| Revision | Date        | Author           | Change Description |
+| -------- | ----------- | ---------------- | ------------------ |
+| 1.0      | Mar 27 2024 | Yaqiang Zhu, Jing Kan | Initial proposal   |
+
+### Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+| ------------------------ | ----------- |
+| FRR | A free and open source Internet routing protocol suite for Linux and Unix platforms |
+| BGP Router ID | 32-bit value that uniquely identifies a BGP device |
+| AS | Autonomous System |
+
+### Scope
+
+This document describes a mechanism to allow user explicitly configure BGP router id.
+
+### Overview
+
+Currently, there are some BGP hard codings in SONiC:
+1. BGP router id was defined as a 32-bit value that uniquely identifies a BGP device. SONiC uses Loopback0 IPv4 address as BGP router id. This coupling prevents users from using customized router id. And FRR would choose an IP address in device to be BGP router id if Loopback0 IPv4 address doesn't exist. If the router id choosen by FRR is not unique in AS, it would be considered an error.
+2. SONiC wouldn't add BGP peer when there is not Loopback0 IPv4 exists. It would cause BGP cannot establish.
+
+Below is current workflow about BGP and router id, only includes contents related to Loopback0.
+
+1. After bgp container started, configuration file `/etc/frr/bgpd.conf` for bgpd would be rendered. It will use Loopback0 IPv4 address as BGP router id, if it doesn't exist, the BGP router id wouldn't be specified.
+2. bgpd start with configuration rendered before. If BGP router id is not specified, it would choose an IP address in device to be BGP router id.
+3. After bgpcfgd started, it will add bgp peer depends on whether Loopback0 IPv4 exist.
+
+<p align=center>
+<img src="img/origin_bgp_seq.png" alt="Figure 1. Origin bgp seq" width=700>
+</p>
+
+### Requirements
+
+Add support to allow user explicitly configure BGP router id.
```

Comment:
In multi-ASIC architectures, inter-ASIC BGP sessions often rely on specific Loopback interfaces or unique IP addresses assigned per ASIC for peering. Could we explicitly address how these scenarios are managed, especially considering that BGP sessions within the device (for the purpose of route exchange between ASICs) and external BGP sessions might have different requirements for identifying and using Router IDs.

Reply:
Updated, currently FrontEnd asic use Loopback4096 as router-id for iBGP and eBGP. BackEnd asic use Loopback4096 as router-id for iBGP. Hence we can specify router-id in config_db of correspond asic